### PR TITLE
Backend/remove username field

### PR DESCRIPTION
**Pull request by wittejm**
This includes changes to endpoint signatures to no longer have a 'username' field. All instances of 'email address' as a json field were replaced with 'email' to sync with other endpoints. 

Changes are reflected in the design doc and test cases. requesting review from front-end folks @hmarcks and @maxkwallace because of API changes.

### CODE REVIEW

**Comment by a-ilango** [comment]
@wittejm thanks for the PR!

I think some of these commits can be squashed together. Can you please squash those that change the same file and/or make sense to be together in a single commit?

**Comment by wittejm** [comment]
@a-ilango Yea I think squashing it all into one makes sense, so I'll do that. I haven't done much group work on git before so wasn't sure that the PR would include all this clutter.

File: doc/design.md [hunk 1]
```diff
@@ -152,17 +151,15 @@ Creates a user
 
 - format: `JSON`
 - fields:
-    * username
-	* encrypted password
-	* email address
+	* email
+    * encrypted password
```
Comment (maxkwallace) [inline doc/design.md]: I don't mean to suggest that this is wrong, honestly just wondering: is there a reason why the indentation is different here for email and encrypted password?

Comment (a-ilango) [reply doc/design.md]: Do you mean before this change? I wonder if spaces and tabs are getting mixed up.

Comment (wittejm) [reply doc/design.md]: It was a tabs and spaces thing. I just fixed those

**Pull request merged**